scripts/west_commands/runners/mdb.py

```diff
@@ -0,0 +1,127 @@
+            cores=args.cores,
+            jtag=args.jtag,
+            nsim_args=args.nsim_args,
+            gui=args.gui,
+            dig_device=args.dig_device
+            )
+
+    def do_run(self, command, **kwargs):
+        self.require(self.commander)
+
+        mdb_basic_options = ['-nooptions', '-nogoifmain',
+                            '-toggle=include_local_symbols=1']
+
+        # nsim
+        if self.nsim_args != '':
+            mdb_target = ['-nsim', '@'+self.nsim_args]
+        # hardware target
+        else:
+            if self.jtag == 'digilent':
+                mdb_target = ['-digilent', self.dig_device]
+            else:
+                mdb_target = ['']
+
+        if command == 'flash':
+            mdb_run = ['-run', '-cl']
+        else:
+            if self.gui:
+                mdb_run = ['-OKN']
+            else:
+                mdb_run = ['-cl']
+
+        if self.cores == '1':
+            mdb_cmd = ([self.commander] + mdb_basic_options + mdb_target +
+                       mdb_run + [self.elf_name])
+        elif self.cores == '2':
+            mdb_sub_cmd1 = ([self.commander] +
+                            ['-pset=1', '-psetname=core0'] +
+                             mdb_basic_options + mdb_target + [self.elf_name])
+            mdb_sub_cmd2 = ([self.commander] +
+                            ['-pset=2', '-psetname=core1', '-prop=download=2'] +
+                             mdb_basic_options + mdb_target + [self.elf_name])
+            mdb_cmd = ([self.commander] +
+                    ['-multifiles=core0,core1']
+                    + mdb_run)
+
+            self.check_call(mdb_sub_cmd1)
+            self.check_call(mdb_sub_cmd2)
+
+        elif self.cores == '4':
+            mdb_sub_cmd1 = ([self.commander] +
+                            ['-pset=1', '-psetname=core0'] +
+                             mdb_basic_options + mdb_target + [self.elf_name])
+            mdb_sub_cmd2 = ([self.commander] +
+                            ['-pset=2', '-psetname=core1', '-prop=download=2'] +
+                             mdb_basic_options + mdb_target + [self.elf_name])
+            mdb_sub_cmd3 = ([self.commander] +
+                            ['-pset=3', '-psetname=core2', '-prop=download=2'] +
+                             mdb_basic_options + mdb_target + [self.elf_name])
+            mdb_sub_cmd4 = ([self.commander] +
+                            ['-pset=4', '-psetname=core3', '-prop=download=2'] +
+                             mdb_basic_options + mdb_target + [self.elf_name])
+            mdb_cmd = ([self.commander] +
+                    ['-multifiles=core0,core1,core2,core3'] +
+                    mdb_run)
+
+            self.check_call(mdb_sub_cmd1)
+            self.check_call(mdb_sub_cmd2)
+            self.check_call(mdb_sub_cmd3)
+            self.check_call(mdb_sub_cmd4)
+
+        else:
+            raise ValueError('unsupported cores {}'.format(self.cores))
+
+        self.check_call(mdb_cmd)
```

Comment:
At some point (either before we start or after we're done) it might make sense to remove `.sc.project` folder which has temporary settings for MDB. This is extremely useful for troubleshooting funny situations with unexpected behavior of the debugger (because it remembers something very special you did in one of the previous debugging sessions and now it gets in the way).

Reply:
@vonhust you don't like that suggestion?

Reply:
I forgot , i will add the code to remove .sc.project